src/expr/src/scalar/func/impls/datum.rs

```diff
@@ -38,7 +38,10 @@ sqlfunc!(
         match a {
             Datum::Null => Ok(None),
             datum => {
-                let sz = mz_repr::datum_size(&datum);
+                // This will undercount sizes for installations that aren't in the
+                // `variable_length_row_encoding` treatment, but
+                // functions need to be determinstic, so there's no better we can do.
+                let sz = mz_repr::datum_size_deterministic::<true>(&datum);
```

Comment:
Why do we decide to undercount rather than overcount? That latter seems safer to me, but that depends on how these SQL functions are used.

Maybe we should overcount as long as the default for the variable length encoding flag is still `false` and switch once we switch the flag default?

Reply:
Btw, if you feel strongly that it's important, I don't mind switching this to the `false` variant